File: cs/algorithms_in/bubble_sort.cs
namespace algorithms_in;

public class BubbleSort
{
    public static void bubbleSort(int[] array) {
        bool needsNextPass = true;
        for (int i = 1; i < array.Length && needsNextPass; i++) {
            needsNextPass = false;
            for (int k = 0; k < array.Length - i; k++) {
                if (array[k] > array[k + 1]) {
                    int temp = array[k];
                    array[k] = array[k + 1];
                    array[k + 1] = temp;
                    needsNextPass = true;
                }
            }
        }
    }
}

File: rust/algorithms_in/src/bubble_sort.rs
pub fn bubble_sort(array: &mut [usize]) {
    let mut needs_next_pass = true;
    let mut i: usize = 1;
    while i < array.len() && needs_next_pass {
        needs_next_pass = false;
        for k in 0..array.len() - i {
            if array[k] > array[k + 1] {
                let temp: usize = array[k];
                array[k] = array[k + 1];
                array[k + 1] = temp;
                needs_next_pass = true;
            }
        }
        i += 1;
    }
}